<launch>
  <env name="ROSCONSOLE_FORMAT" value="[${severity}][${time: %H:%M:%S%F}][${node}]:${message}"/>
  <arg name="obj_num" value="10" />
  <arg name="drone_id" value="0" />

  <arg name="map_size_x" value="20.0" />
  <arg name="map_size_y" value="10.0" />
  <arg name="map_size_z" value=" 2.0" />
  <arg name="odom_topic" value="/mavros/local_position/odom" />
  <arg name="pose_topic" value="/mavros/local_position/pose" />

  <arg name="init_x" value="-10" />
  <arg name="init_y" value="0.0" />
  <arg name="init_z" value="0.5" />
  <arg name="goal_x" value="15" />
  <arg name="goal_y" value="0.0" />
  <arg name="goal_z" value="0.5" />

  <group ns="uav$(arg drone_id)">
    <!-- <node pkg="rast_corridor_planning" name="mapping" type="mapping" output="screen"> -->
    <node pkg="rast_corridor_planning" name="mapping" type="mapping">
      <rosparam file="$(find rast_corridor_planning)/config/cfg.yaml" command="load" />
      <remap from="/camera_front/depth/points" to="pcl_render_node/cloud" />
    </node>

    <node pkg="traj_server" name="traj_server" type="poly_traj_server" output="screen">
      <remap from="~trajectory" to="planning/trajectory" />
      <remap from="/position_cmd" to="planning/pos_cmd" />
      <remap from="~odom" to="$(arg pose_topic)" />
    </node>

    <node pkg="rast_corridor_planning" name="planning" type="multi_planning_node" output="screen">
    <!-- <node pkg="rast_corridor_planning" name="planning" type="multi_planning_node"> -->
      <remap from="~future_risk_full_array" to="mapping/future_risk_full_array" />
      <param name="p_goal_x" value="$(arg goal_x)" />
      <param name="p_goal_y" value="$(arg goal_y)" />
      <param name="p_goal_z" value="$(arg goal_z)" />
      <rosparam file="$(find rast_corridor_planning)/config/cfg.yaml" command="load" />
    </node>

  </group>

  <!-- SIMULATOR -->
  <include file="$(find rast_corridor_planning)/launch/simulator_dyn.launch">
    <arg name="drone_id" value="$(arg drone_id)" />
    <arg name="map_size_x" value="$(arg map_size_x)" />
    <arg name="map_size_y" value="$(arg map_size_y)" />
    <arg name="map_size_z" value="$(arg map_size_z)" />
    <arg name="init_x" value="$(arg init_x)" />
    <arg name="init_y" value="$(arg init_y)" />
    <arg name="init_z" value="$(arg init_z)" />
    <arg name="odometry_topic" value="$(arg odom_topic)" />
    <arg name="odom_pose_topic" value="$(arg pose_topic)" />
  </include>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find rast_corridor_planning)/rviz/sim.rviz" >
    <remap from="/move_base_simple/goal" to="$(arg pose_topic)" />
  </node>
  <!--	<node pkg="rviz" type="rviz" name="rviz2" args="-d $(find dynamic_occpuancy_map)/rviz/future_status.rviz"/>-->
</launch>